### PR TITLE
feat(value): add null

### PR DESCRIPTION
**Pull request by tony-go**

### CODE REVIEW

File: src/engine/include/includejs/engine_value.h [hunk 1]
```diff
@@ -34,6 +34,7 @@ class INCLUDEJS_ENGINE_EXPORT Value {
   auto is_error() const -> bool;
   auto is_object() const -> bool;
   auto is_boolean() const -> bool;
+  auto is_null() const -> bool;
```
Comment (jviotti) [inline src/engine/include/includejs/engine_value.h:37]: This one is good. We missed this

File: src/engine/javascript_core/engine_context.cc [hunk 2]
```diff
@@ -29,6 +29,10 @@ auto Context::make_undefined() const -> Value {
           JSValueMakeUndefined(this->internal->context)};
 }
 
+auto Context::make_null() const -> Value {
+  return {this->internal->context, JSValueMakeNull(this->internal->context)};
```
Comment (jviotti) [inline src/engine/javascript_core/engine_context.cc:33]: We can change it to this, and move it to `src/engine/common`, as its no longer JavaScriptCore specific

```suggestion
  return from(nullptr);
```

File: src/engine/include/includejs/engine_context.h [hunk 3]
```diff
@@ -21,6 +21,7 @@ class INCLUDEJS_ENGINE_EXPORT Context {
   ~Context();
 #endif
   auto make_undefined() const -> Value;
+  auto make_null() const -> Value;
```
[the remark this answers is not in the document]
Comment (tony-go) [reply src/engine/include/includejs/engine_context.h:24]: I just miss this from(std::nullptr_t value)

I think we should keep the conservation of mechanism as much as we could, I will simply remove it and use .from

[the remark this answers is not in the document]
Comment (tony-go) [reply src/engine/include/includejs/engine_context.h:24]: > I deleted make_null on purpose,

Sorry I think it was simply missing I did not realized we removed it

[the remark this answers is not in the document]
Comment (tony-go) [reply src/engine/include/includejs/engine_context.h:24]: The function was not implemented tho :)

**Review by jviotti — APPROVED** [review]

File: src/engine/include/includejs/engine_context.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (jviotti) [reply src/engine/include/includejs/engine_context.h]: Ah, or maybe I'm misremembering. Not sure :)

**Pull request merged**